src/proxyConnect.ts

```diff
@@ -0,0 +1,105 @@
+import { Socket } from 'net'
+import { DOMParser } from 'xmldom'
+import { EventEmitter } from 'events'
+import { decode } from 'iconv-lite'
+import { ENCODING } from './xdebugConnection'
+
+export const DEFAULTIDEKEY = 'vsc'
+
+/** Informs proxy of incoming connection and who to pass data back to. */
+export class ProxyConnect extends EventEmitter {
+    /** Port editor is listening on (default: 9001 */
+    private _port: number
+    /** a CLI binary boolean option (default: 1) */
+    private _allowMultipleSessions: number
+    /** host domain or ip (default: 127.0.0.1) */
+    private _host: string
+    /** unique key that allows the proxy to match requests to your editor. (default: DEFAULTIDEKEY) */
+    private _key: string
+    /** proxy response data parser */
+    private _parser = new DOMParser()
+    /** tcp connection to communicate with proxy server */
+    public _socket = new Socket()
```

Comment:
Why is this public?

Reply:
IIRC, when we began writing our tests we mocked out the socket and such but weren't completely successful. We opted to make the socket public and that allowed us to test each step.

Reply:
The way the test uses this should-be private property to assert things looks very hacky to me, as it is making use of internals and testing implementation details. I am not confident I would be able to maintain and support this code if issues pop up. Please see if you can find a way to define clear interface boundaries and to test at those boundaries.

Reply:
@felixfbecker I changed it back to `private` and followed the convention you used to pass in the socket in the constructor. Please let me know if this is acceptable.